### PR TITLE
Implement edit function

### LINKED ISSUE #106: Feature to edit entries
As a careless user, I want to be able to edit my entries, so that I don't have to delete and add an incorrectly keyed entry

### PR DESCRIPTION
**Pull request by ChongQiRong**
Fixes #106, #145

### CODE REVIEW

**Review by lil1n — APPROVED** [review]
lgtm

File: src/main/java/seedu/rainyDay/command/EditCommand.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        setupLogger();
+        logger.log(Level.INFO, "starting EditCommand.execute()");
+
+        index -= 1;
+
+        int previousStatementCount = financialReport.getStatementCount();
+        assert (index < financialReport.getStatementCount() && index >= 0) : "invalid index provided for edit";
+
+        financialReport.deleteStatement(index);
+        financialReport.addStatementAtIndex(new FinancialStatement(description, flowDirection, value, category), index);
```
Comment (azriellee) [inline src/main/java/seedu/rainyDay/command/EditCommand.java:50]: Would a more direct solution be to directly amend the statement by getting the financialStatement object using financialReport.get(index) and amend it directly? This way we can also allow users to amend particular fields only

**Review by azriellee — COMMENTED** [review]
Looks ok! Can consider my suggestions for EditCommand.java, thank you!

File: src/main/java/seedu/rainyDay/exceptions/ErrorMessage.java [hunk 2]
```diff
@@ -7,7 +7,9 @@ public enum ErrorMessage {
     NO_DELETE_INDEX("Please include a delete index!"),
     WRONG_ADD_FORMAT("'add' input not correct!\nPlease refer to 'help' for correct user inputs for 'add' commands"),
     WRONG_FILTER_FORMAT("Please refer to 'help' for correct user input!"),
-    NO_STATEMENTS_MATCH_DESCRIPTION("We could not find any matches for your description in your report");
+    NO_STATEMENTS_MATCH_DESCRIPTION("We could not find any matches for your description in your report"),
+    WRONG_EDIT_INDEX("Please ensure edit index is a number!"),
+    NO_EDIT_INDEX("Please include a  edit index!");
```
Comment (KN-CY) [inline src/main/java/seedu/rainyDay/exceptions/ErrorMessage.java:12]: Double space in the error message

File: src/main/java/seedu/rainyDay/command/EditCommand.java (hunk 1 above)
Comment (ChongQiRong) [reply src/main/java/seedu/rainyDay/command/EditCommand.java:50]: Totally agreed

File: src/main/java/seedu/rainyDay/exceptions/ErrorMessage.java (hunk 2 above)
Comment (ChongQiRong) [reply src/main/java/seedu/rainyDay/exceptions/ErrorMessage.java:12]: Amended

File: src/main/java/seedu/rainyDay/command/EditCommand.java [hunk 3]
```diff
@@ -46,10 +67,22 @@ public CommandResult execute() {
         int previousStatementCount = financialReport.getStatementCount();
         assert (index < financialReport.getStatementCount() && index >= 0) : "invalid index provided for edit";
 
-        financialReport.deleteStatement(index);
-        financialReport.addStatementAtIndex(new FinancialStatement(description, flowDirection, value, category), index);
+        if (flag == "") {
```
Comment (lil1n) [inline src/main/java/seedu/rainyDay/command/EditCommand.java:70]: could consider flag.isEmpty()

**Review by lil1n — COMMENTED** [review]
looks good

File: src/main/java/seedu/rainyDay/command/EditCommand.java (hunk 3 above)
Comment (ChongQiRong) [reply src/main/java/seedu/rainyDay/command/EditCommand.java]: Ok

**Pull request merged**